instrumentation/opentelemetry-instrumentation-pika/src/opentelemetry/instrumentation/pika/utils.py

```diff
@@ -46,17 +46,18 @@ def decorated_callback(
         ctx = propagate.extract(properties.headers, getter=_pika_getter)
         if not ctx:
             ctx = context.get_current()
+        token = context.attach(ctx)
         span = _get_span(
             tracer,
             channel,
             properties,
             span_kind=SpanKind.CONSUMER,
             task_name=task_name,
-            ctx=ctx,
             operation=MessagingOperationValues.RECEIVE,
         )
         with trace.use_span(span, end_on_exit=True):
             retval = callback(channel, method, properties, body)
+        context.detach(token)
```

Comment:
If the callback raises, the context won't detach. Change to:
```python
try:
    with trace.use_span(span, end_on_exit=True):
        retval = callback(channel, method, properties, body)
finally:
    context.detach(token)
```

Reply:
Fixed